src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -0,0 +1,201 @@
+        /// </summary>
+        /// <param name="sc">Spark Context object</param>
+        /// <param name="path">Path where file is to be created</param>
+        /// <param name="value">Broadcast value to be written to the file</param>
+        private void WriteBroadcastValueToFile(SparkContext sc, string path, object value)
+        {
+            Directory.CreateDirectory(sc.TempDir);
+            using FileStream f = File.Create(path);
+            Dump(value, f);
+        }
+
+        /// <summary>
+        /// Function that serializes and stores the object passed to the given Stream.
+        /// </summary>
+        /// <param name="value">Serializable object</param>
+        /// <param name="stream">Stream to which the object is serialized</param>
+        private void Dump(object value, Stream stream)
+        {
+            var formatter = new BinaryFormatter();
+            formatter.Serialize(stream, value);
+        }
+    }
+
+    /// <summary>
+    /// Global registry to store the value of all active broadcast variables. It is populated on
+    /// the worker when accessed through BroadcastVariableProcessor. The value is returned whenever
+    /// the user invokes Broadcast.Value().
+    /// </summary>
+    internal static class BroadcastRegistry
+    {
+        public static ConcurrentDictionary<long, object> s_registry = 
+            new ConcurrentDictionary<long, object>();
+
+        /// <summary>
+        /// Function to add the value of the broadcast variable to s_registry.
+        /// </summary>
+        /// <param name="bid">Id of the Broadcast variable object to add</param>
+        /// <param name="value">Value of the Broadcast variable</param>
+        public static void Add(long bid, object value)
+        {
+            s_registry.TryAdd(bid, value);
+        }
+
+        /// <summary>
+        /// Function to remove the Broadcast variable from s_regitry.
+        /// </summary>
+        /// <param name="bid">Id of the Broadcast variable object to remove</param>
+        public static void Remove(long bid)
+        {
+            s_registry.TryRemove(bid, out _);
+        }
+    }
+
+    /// <summary>
+    /// Stores the JVMObjectReference objects of all active broadcast variables that
+    /// are sent to each executor through the CreatePythonFunction.
+    /// </summary>
+    internal static class JvmBroadcastRegistry
+    {
+        public static List<JvmObjectReference> s_jvmBroadcastVariables = 
+            new List<JvmObjectReference>();
+
+        /// <summary>
+        /// Converts Generic.List of JvmObjectReference to ArrayList of JvmObjectReference
+        /// </summary>
+        /// <param name="jvm">JVM bridge to use</param>
+        /// <returns>ArrayList object containing JvmObjectReference objects for all
+        /// broadcast variables</returns>
+        public static ArrayList ConvertToArrayList(IJvmBridge jvm)
```

Comment:
Can we move this logic to the caller side? And you can expose `GetAll()` in this class.